test2code/src/test/kotlin/com/epam/drill/test2code/coverage/InMemoryRetentionQueueTest.kt

```diff
@@ -0,0 +1,163 @@
+package com.epam.drill.test2code.coverage
+
+import org.junit.jupiter.api.*
+import java.math.*
+
+class InMemoryRetentionQueueTest {
+
+    @Test
+    fun `adding multiple elements exceeding limit should maintain total size at limit`() {
+        val byteCount = 6
+        val innerArraySize = 2
+        val bytes = Array(byteCount) { ByteArray(innerArraySize) { 1 } }.asSequence()
+        val sizeLimit = BigInteger.valueOf(10)
+        val queue = InMemoryRetentionQueue(totalSizeByteLimit = sizeLimit)
+
+        queue.addAll(bytes)
+
+        val actualSizeOfQueue = queue.flush().sumOf { it.size }
+        Assertions.assertEquals(sizeLimit.toInt(), actualSizeOfQueue)
+    }
+
+    @Test
+    fun `adding elements with size equal to limit should result in queue size exactly at limit`() {
+        val byteCount = 1
+        val innerArraySize = 10
+        val bytes = Array(byteCount) { ByteArray(innerArraySize) { 1 } }.asSequence()
+        val sizeLimit = BigInteger.valueOf(10)
+        val queue = InMemoryRetentionQueue(totalSizeByteLimit = sizeLimit)
+
+        queue.addAll(bytes)
+
+        val actualSizeOfQueue = queue.flush().sumOf { it.size }
+        Assertions.assertEquals(sizeLimit.toInt(), actualSizeOfQueue)
+    }
+
+    @Test
+    fun `adding single element larger than limit should result in an empty queue`() {
+        val queue = InMemoryRetentionQueue(totalSizeByteLimit = BigInteger.valueOf(10))
+        val byteCount = 1
+        val innerArraySize = 12
+        val bytes = Array(byteCount) { ByteArray(innerArraySize) { 1 } }.asSequence()
+
+        queue.addAll(bytes)
+
+        Assertions.assertTrue(queue.flush().count() == 0)
+    }
+
+    @Test
+    fun `flush should return the same elements as added to the queue`() {
+        val queue = InMemoryRetentionQueue(totalSizeByteLimit = BigInteger.valueOf(10))
+        val byteCount = 1
+        val innerArraySize = 5
+        val bytes = Array(byteCount) { ByteArray(innerArraySize) { 1 } }.asSequence()
+
+        queue.addAll(bytes)
+        val actualBytes = queue.flush()
+
+        Assertions.assertEquals(bytes.toList(), actualBytes.toList())
+    }
+
+    @Test
+    fun `queue after flush must be empty`() {
+        val queue = InMemoryRetentionQueue(totalSizeByteLimit = BigInteger.valueOf(10))
+        val byteCount = 1
+        val innerArraySize = 5
+        val bytes = Array(byteCount) { ByteArray(innerArraySize) { 1 } }.asSequence()
+
+        queue.addAll(bytes)
+
+        val bytes1 = queue.flush().toList()
+        val bytes2 = queue.flush().toList()
+
+        Assertions.assertTrue(bytes2.isEmpty())
+    }
+
+    @Test
+    fun `queue should only fit elements with size less than or equal to total limit`() {
```

Comment:
Edit test name to reflect test purpose

Reply:
"after attempt to add element over limit queue should still accept elements up to limit"

Reply:
done